samples/basic-approval-workflow/src/pubnub-client.js

```diff
@@ -0,0 +1,85 @@
+const channelName = (prefix, entrySys) => [
+  prefix,
+  entrySys.space.sys.id,
+  entrySys.environment.sys.id,
+  entrySys.id,
+].join('!');
+
+export default async function createPubNubClient({
+  PubNub,
+  publishKey,
+  subscribeKey,
+  channelPrefix,
+  entrySys,
+  onChange,
+}) {
+  const pubnub = new PubNub({ publishKey, subscribeKey, ssl: true });
+  const channel = channelName(channelPrefix, entrySys);
+  const state = { log: [] };
+
+  function addToLog({ entry, timetoken }) {
+    const validEntry = entry && typeof entry.id === 'string';
+    const validTimetoken = typeof timetoken === 'string';
+    const newEntry = !state.log.find(cur => cur.id === entry.id);
+
+
+    if (validEntry && validTimetoken && newEntry) {
+      // Timetoken is a 17-digit precision unix time (UTC)
+      const t = parseInt(timetoken, 10) / 10000;
```

Comment:
`Number.MAX_SAFE_INTEGER` is lower than a 17 digit unix time (it has only 16 digits).
As we probably don't want to use `BigInt` here I'd just slice away the last four digits.